src/Features/Core/Portable/NameTupleElement/AbstractNameTupleElementCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,110 @@
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.NameTupleElement
+{
+    abstract class AbstractNameTupleElementCodeRefactoringProvider<TArgumentSyntax, TTupleExpressionSyntax> : CodeRefactoringProvider
+        where TArgumentSyntax : SyntaxNode
+        where TTupleExpressionSyntax : SyntaxNode
+    {
+        protected abstract bool IsCloseParenOrComma(SyntaxToken token);
+        protected abstract TArgumentSyntax WithName(TArgumentSyntax argument, string argumentName);
+        protected abstract bool IsUnnamedElement(TArgumentSyntax node);
+        protected abstract SeparatedSyntaxList<TArgumentSyntax> GetArguments(TTupleExpressionSyntax tuple);
+
+        public override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
+        {
+            var document = context.Document;
+            if (document.Project.Solution.Workspace.Kind == WorkspaceKind.MiscellaneousFiles)
+            {
+                return;
+            }
+
+            if (context.Span.Length > 0)
+            {
+                return;
+            }
+
+            var cancellationToken = context.CancellationToken;
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var position = context.Span.Start;
+            var token = root.FindToken(position);
+            if (token.Span.Start == position &&
+                IsCloseParenOrComma(token))
+            {
+                token = token.GetPreviousToken();
+                if (token.Span.End != position)
+                {
+                    return;
+                }
+            }
+
+            var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
+            var argument = root.FindNode(token.Span)
+                .GetAncestorsOrThis<TArgumentSyntax>()
+                .FirstOrDefault(node => syntaxFacts.IsTupleExpression(node.Parent));
+
+            if (argument == null || !IsUnnamedElement(argument))
+            {
+                return;
+            }
+
+            var tuple = (TTupleExpressionSyntax)argument.Parent;
+
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var tupleType = semanticModel.GetTypeInfo(tuple, cancellationToken).ConvertedType as INamedTypeSymbol;
+            if (tupleType == null)
+            {
+                return;
+            }
+
+            var arguments = GetArguments(tuple);
+            var argumentIndex = arguments.IndexOf(argument);
+            var elements = tupleType.TupleElements;
+            if (elements.IsDefaultOrEmpty || argumentIndex >= elements.Length)
+            {
+                return;
+            }
+
+            var element = elements[argumentIndex];
+            if ((object)element == element.CorrespondingTupleField)
```

Comment:
shouldn't this just be .Equals?

Reply:
When you don't name a tuple element, we'll return the default field from `.TupleElements`. So if the `.CorrespondingTupleField` for a tuple element is itself, then we have a default field (rather than a named element).